dlib/test/vectorstream.cpp

```diff
@@ -59,7 +60,7 @@ namespace
         DLIB_TEST(s.get() == EOF);
 
         s.clear();
-        s.seekg(6);
+        s.seekg(6, std::ios_base::beg);
```

Comment:
I think `seekg(6)` does (or at least can) execute a different codepath through the iostreams than `seekg(6, beg)`.  So this line should remain `seekg(6)`.  Then later on you should call the 3 other variants of `seekg()` and verify that the right thing happened.